### PR TITLE
PIM-1016: PIM | Export | Unable to process PLP All Attributes export

### PR DESCRIPTION
**Pull request by danielxji6**

### CODE REVIEW

File: legacy/PimExportHelper.js [hunk 1]
```diff
@@ -43,15 +43,21 @@ class PimExportHelper {
    * @param {string} fieldApi
    */
   getValue(object, fieldApi) {
+    if (!object || !fieldApi) {
+      return null
+    }
     let fields = fieldApi.split('.');
     let queryResult = object;
-    fields.forEach(field => {
+    for (const field of fields) {
```
Comment (mfullmore) [inline legacy/PimExportHelper.js:51]: why are you moving away from the forEach?

File: legacy/PimExportHelper.js [hunk 2]
```diff
@@ -43,15 +43,21 @@ class PimExportHelper {
...
+      if (!queryResult) {
```
Comment (mfullmore) [inline legacy/PimExportHelper.js:52]: not sure this is needed, you already return null if `object` is null

File: legacy/PimExportHelper.js [hunk 3]
```diff
@@ -43,15 +43,21 @@ class PimExportHelper {
    * @param {string} fieldApi
    */
   getValue(object, fieldApi) {
+    if (!object || !fieldApi) {
```
Comment (mfullmore) [inline legacy/PimExportHelper.js:46]: I would change this to
```
if (object === null || fieldApi === null)...
```
this covers empty as well as undefined.

Comment (danielxji6) [reply legacy/PimExportHelper.js:46]: not following. We do want to check empty or undefined, right?

File: legacy/PimExportHelper.js (hunk 2 above)
Comment (danielxji6) [reply legacy/PimExportHelper.js:52]: So i can use `break` to prevent null pointer

File: legacy/PimExportHelper.js (hunk 1 above)
Comment (danielxji6) [reply legacy/PimExportHelper.js:51]: So i can use break to prevent null pointer

File: legacy/PimExportHelper.js (hunk 3 above)
Comment (mfullmore) [reply legacy/PimExportHelper.js:46]: never mind I said this comment.

File: legacy/PimExportHelper.js (hunk 1 above)
Comment (mfullmore) [reply legacy/PimExportHelper.js:51]: and you can just check if it's there and then process. 
```
fields.forEach((field) => {
  if (queryResult) {
    if (field.endsWith('__c') || field.endsWith('__r')) {
        queryResult = queryResult[this.namespace(field)];
      } else {
        queryResult = queryResult.field;
      }
  }
});
```
But what ever is clever.

File: legacy/PimExportHelper.js (hunk 2 above)
Comment (mfullmore) [reply legacy/PimExportHelper.js:52]: but the variable `queryResult` doesn't change when you iterate through the array. Unless I am missing something.

Comment (mfullmore) [reply legacy/PimExportHelper.js:52]: never mind on that one as well. I saw that queryResult does change later in the logic.

**Review by mfullmore — APPROVED** [review]

**Pull request merged**